### PR TITLE
S3 bucket deployment

### PR DESCRIPTION
**Pull request by AlphabeticalAnomaly**

### CODE REVIEW

File: dailymail/dailymail_stack.py [hunk 1]
```diff
@@ -7,38 +7,40 @@
...
+        email = CfnParameter(self, "email", type="string", description="email address")
+        scheduled_lambda = lambda_.Function(self, "MyDailyMail",
                                             handler='lambda_function.lambda_handler',
                                             runtime=lambda_.Runtime.PYTHON_3_8,
                                             code=lambda_.Code.from_asset('lambda'),
                                             )
+        bucket = aws_s3.Bucket(self, "TestBucket", bucket_name="testbucketcdk1241210",)
+        bucket.grant_read(scheduled_lambda)
+        deployment = aws_s3_deployment.BucketDeployment(self, "TestDeployment",
+                                                        sources=[aws_s3_deployment.Source.asset(path="../DAILYMAIL/resource")],
```
Comment (EduardDranca) [inline dailymail/dailymail_stack.py]: This path looks weird, why the capitalization and the `../`?

File: dailymail/dailymail_stack.py [hunk 2]
```diff
@@ -7,38 +7,40 @@
...
         super().__init__(scope, construct_id, **kwargs)
-
-        scheduled_lambda = lambda_.Function(self, "MyDailyMailTest",
+        email = CfnParameter(self, "email", type="string", description="email address")
+        scheduled_lambda = lambda_.Function(self, "MyDailyMail",
                                             handler='lambda_function.lambda_handler',
                                             runtime=lambda_.Runtime.PYTHON_3_8,
                                             code=lambda_.Code.from_asset('lambda'),
                                             )
+        bucket = aws_s3.Bucket(self, "TestBucket", bucket_name="testbucketcdk1241210",)
```
Comment (EduardDranca) [inline dailymail/dailymail_stack.py:26]: I see the bucket_name is reused bellow. Either create a constant for it or reference it from the bucket object when you use it on line 45.

File: lambda/lambda_function.py [hunk 3]
```diff
@@ -1,26 +1,35 @@
...
 
 def lambda_handler(event, context):
-    content = event["content"]
+    bucket = event["bucket"]
     address = event["address"]
-    source = event["address"]
+    content = event["content"]
+    response_address = s3.get_object(Bucket=bucket, Key=address)
+    address_data = response_address['Body'].read()
+    address_str = address_data.decode('utf-8')
```
Comment (EduardDranca) [inline lambda/lambda_function.py]: Create a function for this

File: resource/arn.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+arn:aws:ses:eu-north-1:916886530732:identity/rapanosul69@gmail.com
```
Comment (EduardDranca) [inline resource/arn.txt]: I don't think you need this anymore.

File: dailymail/dailymail_stack.py [hunk 5]
```diff
@@ -7,38 +7,40 @@
...
 
         rule.add_target(aws_events_targets.LambdaFunction(scheduled_lambda, event=events.RuleTargetInput.from_object(
-            {"content": "Your daily CDK", "address": "example@domain.com"}), retry_attempts=1))
-
-        bucket = aws_s3.Bucket(self, "TestBucket", bucket_name="testbucketcdk12412162234155")
-
-
-
+            {"bucket": "testbucketcdk1241210", "address": f"{email}", "content": "email_content.txt",
+             "arn": "arn.txt"}), retry_attempts=1))
```
Comment (EduardDranca) [inline dailymail/dailymail_stack.py]: You don't use the arn in the handler anymore, you can remove it from the event.

File: dailymail/dailymail_stack.py [hunk 6]
```diff
@@ -7,38 +7,39 @@
...
+        email = CfnParameter(self, "email", type="string", description="email address")
+        scheduled_lambda = lambda_.Function(self, "MyDailyMail",
                                             handler='lambda_function.lambda_handler',
                                             runtime=lambda_.Runtime.PYTHON_3_8,
                                             code=lambda_.Code.from_asset('lambda'),
                                             )
+        bucket = aws_s3.Bucket(self, "TestBucket", bucket_name="testbucketcdk1241210",)
+        bucket.grant_read(scheduled_lambda)
+        deployment = aws_s3_deployment.BucketDeployment(self, "TestDeployment",
+                                                        sources=[aws_s3_deployment.Source.asset(path="DailyMail/resource")],
```
Comment (EduardDranca) [inline dailymail/dailymail_stack.py]: Does cdk deploy work with this path?

File: lambda/decode_function.py [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import boto3
+
+
+def object_body_to_str(bucket_name=str, object_key=str):
+    s3 = boto3.client('s3')
```
Comment (EduardDranca) [inline lambda/decode_function.py]: The s3 client should only be initialized once, not every time you want to read a file, you should create a class for this method.

Comment (EduardDranca) [reply lambda/decode_function.py]: Also, function names with the form x_to_y are usually mapping functions, where you pass a parameter of type x and map it to an object of type y. A better name for this function would just be `read_object_content`.

**Review by EduardDranca — APPROVED** [review]

**Comment by AlphabeticalAnomaly** [comment]
S3_bucket_deployment branch merged with the master branch.

**Pull request merged**